### ISSUE TITLE
Complete FFI implementation with proper embed and logits support

### ISSUE THREAD

**Issue by EffortlessSteven** [EffortlessMetrics/BitNet-rs#86]
## Summary
The FFI (Foreign Function Interface) implementation has stub functions using `todo!()` for critical functionality, breaking C API compatibility.

## Current Issue
In `crates/bitnet-ffi/src/inference.rs`:
```rust
// Lines 385-393
todo!(\"embed not used in bitnet-ffi tests\")
todo!(\"logits not used in bitnet-ffi tests\")
```

## Impact
- **API Incompleteness**: C API users cannot access embeddings or logits
- **Runtime Panics**: Calling these functions causes immediate panic
- **Compatibility Breaking**: Violates llama.cpp drop-in replacement promise
- **Production Blocking**: Cannot be used in production with these stubs

### Embed Function
- Implement text/token embedding extraction
- Return embeddings in C-compatible format
- Handle memory management for returned embeddings
- Support batch embedding operations

### Logits Function
- Expose raw model logits for external processing
- Implement proper memory layout for C consumption
- Support both single and batch logit extraction
- Handle vocabulary size variations

## Technical Requirements
1. **Memory Safety**: Proper C-compatible memory management
2. **Error Handling**: Graceful failure without panics
3. **Performance**: Efficient data copying/sharing with C
4. **Compatibility**: Match llama.cpp API signatures exactly

## Dependencies
- Requires inference engine to expose embeddings/logits APIs
- May need updates to core BitNet model interface
- Needs comprehensive C API test coverage

## Testing Requirements
- [ ] C API integration tests for embed functionality
- [ ] C API integration tests for logits extraction
- [ ] Memory leak detection for C API boundary
- [ ] Performance benchmarks vs llama.cpp equivalents
- [ ] Cross-platform compatibility testing

## Acceptance Criteria
- [ ] Remove all `todo!()` stubs from FFI implementation
- [ ] Implement complete embed and logits functionality
- [ ] Add comprehensive C API tests
- [ ] Update C API documentation
- [ ] Verify drop-in compatibility with llama.cpp
- [ ] Performance parity with reference implementation

## Priority
**HIGH** - Blocks C API production readiness

## Related Components
- Core inference engine API
- C FFI bindings and memory management
- Cross-validation test suite
- Python bindings (may also be affected)
